### PR TITLE
#52 add team and league models

### PR DESCRIPTION
**Pull request by nakometa**
This task is related to Issue #52 . The current models have the following connections:
One Sport -> Many Leagues.
One League -> Many Teams & SubLeagues.

Let me know if you want to change the current model or if something is wrong.

### CODE REVIEW

File: SportsHub/SportsHub.DAL/Data/Configurations/SportEntityTypeConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+using SportsHub.DAL.Data.Configurations.Constants;
+using SportsHub.Domain.Models;
+
+namespace SportsHub.DAL.Data.Configurations
+{
+    public class SportEntityTypeConfiguration : IEntityTypeConfiguration<Sport>
+    {
+        public void Configure(EntityTypeBuilder<Sport> sport)
+        {
+            // TODO: CHANGE ALL CONSTANTS
```
Comment (Banovvv) [inline SportsHub/SportsHub.DAL/Data/Configurations/SportEntityTypeConfiguration.cs:12]: Constants seem to be updated, if so - comment can be removed

File: SportsHub/SportsHub.Core/Models/League.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿namespace SportsHub.Domain.Models
+{
+    public class League
+    {
+        public League()
+        {
+            Teams = new HashSet<Team>();
```
Comment (Banovvv) [inline SportsHub/SportsHub.Core/Models/League.cs:7]: We have been asked to remove `ICollection` initializations from Model's constructors before, maybe consider doing the same here for consistency?

**Review by Banovvv — CHANGES_REQUESTED** [review]

**Review by Banovvv — APPROVED** [review]

**Review by MBocharskyi — APPROVED** [review]

File: SportsHub/SportsHub.DAL/Data/Configurations/Constants/ConfigurationConstants.cs [hunk 3]
```diff
@@ -2,6 +2,14 @@
 {
     public static class ConfigurationConstants
     {
+        public const int SportNameMaxLength = 50;
```
Comment (NataliiaBidiuk) [inline SportsHub/SportsHub.DAL/Data/Configurations/Constants/ConfigurationConstants.cs:5]: lets create separate classes inside of ConfigurationConstants for every entity

File: SportsHub/SportsHub.DAL/Data/Configurations/SportEntityTypeConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+using SportsHub.Domain.Models;
+
+namespace SportsHub.DAL.Data.Configurations
+{
+    public class SportEntityTypeConfiguration : IEntityTypeConfiguration<Sport>
+    {
+        public void Configure(EntityTypeBuilder<Sport> sport)
+        {
+            sport.Property(x => x.Name)
+                .IsRequired(true)
```
Comment (NataliiaBidiuk) [inline SportsHub/SportsHub.DAL/Data/Configurations/SportEntityTypeConfiguration.cs:13]: I suppose that name may be Unique here?

Comment (NataliiaBidiuk) [reply SportsHub/SportsHub.DAL/Data/Configurations/SportEntityTypeConfiguration.cs:13]: I think that all these tables should have unique names, am I right?

File: SportsHub/SportsHub.DAL/Data/Configurations/Constants/ConfigurationConstants.cs (hunk 3 above)
Comment (nakometa) [reply SportsHub/SportsHub.DAL/Data/Configurations/Constants/ConfigurationConstants.cs:5]: Do you mean creating separate files containing the constants for each class and then injecting them into ConfigurationConstants or just creating the files without injecting them?

Comment (NataliiaBidiuk) [reply SportsHub/SportsHub.DAL/Data/Configurations/Constants/ConfigurationConstants.cs:5]: We can have class inside of class.
 static class ConfigurationConstants{
Team{
constans
}
User{
}
...
}

**Pull request merged**